pdns/dnsdistdist/dnsdist-discovery.cc

```diff
@@ -272,8 +272,9 @@ bool ServiceDiscovery::getDiscoveredConfig(const UpgradeableBackend& upgradeable
 
     sock.writenWithTimeout(reinterpret_cast<const char*>(packet.data()), packet.size(), backend->d_config.tcpSendTimeout);
 
+    const struct timeval remainingTime = {.tv_sec = backend->d_config.tcpRecvTimeout, .tv_usec = 0};
     uint16_t responseSize = 0;
-    auto got = sock.readWithTimeout(reinterpret_cast<char*>(&responseSize), sizeof(responseSize), backend->d_config.tcpRecvTimeout);
+    auto got = readn2WithTimeout(sock.getHandle(), &responseSize, sizeof(responseSize), remainingTime);
     if (got < 0 || static_cast<size_t>(got) != sizeof(responseSize)) {
```

Comment:
```suggestion
    if (got < 0 || static_cast<size_t>(got) != sizeof(responseSize)) {
```
as `readn2WithTimeout` returns  `size_t` this test is redundant.

Reply:
#13138 :)

Reply:
Right, I'll include the commits from the second follow-up PR in this one :)

Reply:
Done!